redbot/cogs/warnings/warnings.py

```diff
@@ -76,6 +84,54 @@ def __init__(self, bot: Red):
         else:
             await ctx.send(_("Custom reasons have been disabled."))
 
+    @warningset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
+        """Toggle whether warnings should be sent to users in DMs."""
+        guild = ctx.guild
+        toggle = not await self.config.guild(guild).toggle_dm()
+        await self.config.guild(guild).toggle_dm.set(toggle)
+        if toggle:
+            await ctx.send(_("I will now try to send warnings to users DMs."))
+        else:
+            await ctx.send(_("Warnings will no longer be sent to users DMs."))
+
+    @warningset.command()
+    @commands.guild_only()
+    async def warnchannel(self, ctx: commands.Context, channel: discord.TextChannel = None):
+        """Set the channel where warnings should be sent to.
+
+        Leave empty to use the channel `[p]warn` command was called in.
+        """
+        guild = ctx.guild
+        if channel:
+            await self.config.guild(guild).warn_channel.set(channel.id)
+            await ctx.send(
+                _("The warn channel has been set to {channel}.").format(channel=channel.mention)
+            )
+        else:
+            await self.config.guild(guild).warn_channel.set(channel)
+            await ctx.send(_("Warnings will now be sent in the channel command was used in."))
+
+    @warningset.command()
+    @commands.guild_only()
+    async def togglechannel(self, ctx: commands.Context):
+        """
+        Toggle if warnings should be sent to a channel set with `[p]warningset warnchannel`.
+        """
+        guild = ctx.guild
+        toggle = await self.config.guild(guild).toggle_channel()
+        await self.config.guild(guild).toggle_channel.set(not toggle)
+        channel = await self.config.guild(guild).warn_channel()
+        if not toggle:
+            await ctx.send(
+                _("Warnings will now be sent to {channel}.").format(channel=channel.mention)
+            )
+        else:
+            await ctx.send(
+                _("Warns will no longer be sent to {channel}.").format(channel=channel.mention)
+            )
```

Comment:
`channel.mention` won't work here because:
a) `channel` variable isn't a channel object
b) warn channel might not be set at all - bot should then say it will send warnings in the channel command was used in